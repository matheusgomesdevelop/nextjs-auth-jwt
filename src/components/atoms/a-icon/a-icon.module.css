.a_icon {
  font-size: var(--fs-lg);
}
