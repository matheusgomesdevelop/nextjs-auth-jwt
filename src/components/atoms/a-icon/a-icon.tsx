import styles from "./a-icon.module.css";

export type T_AIconList = "pi-lock" | "pi-github" | "pi-linkedin" | "pi-sun" | "pi-spin pi-cog";

interface I_AIcon {
  icon: T_AIconList;
}

export const A_Icon = ({ icon }: I_AIcon) => {
  return <i className={`${styles.a_icon} pi ${icon}`}></i>;
};
