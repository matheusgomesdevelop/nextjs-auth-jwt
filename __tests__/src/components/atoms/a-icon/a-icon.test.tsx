import renderer from "react-test-renderer";

// Atoms
import { A_Icon } from "@/components/atoms/a-icon/a-icon";

describe("Deve renderizar os icones corretamente", () => {
  it("Deve preservar a estrutura visual do icone", () => {
    const three = renderer.create(<A_Icon icon="pi-lock" />).toJSON();

    expect(three).toMatchSnapshot();
  });
});
