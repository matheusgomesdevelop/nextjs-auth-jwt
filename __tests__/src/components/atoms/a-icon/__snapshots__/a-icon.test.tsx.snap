// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar os icones corretamente Deve preservar a estrutura visual do icone 1`] = `
<i
  className="a_icon pi pi-lock"
/>
`;
